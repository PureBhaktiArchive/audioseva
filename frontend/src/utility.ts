import _ from "lodash";
import moment from "moment";
import { ICount } from "@/types/DataTable";

export const filteredStatus = ["Lost", "Opted out", "Incorrect", "Duplicate"];

export function getDayDifference(date: number) {
  return moment().diff(date, "days");
}

type MomentDateOrString = string|moment.Moment

export const getLastDays = (day = 5) => {
  const days: MomentDateOrString[] = ["today"];
  const today = moment();
  let i;
  for (i = 1; i < day + 1; i++) {
    days.unshift(today.clone().subtract(i, "days"))
  }
  return days;
};

export const mergeDoneStatistics = (doneStatistics: ICount) => {
  const baseStatistics: ICount = {};
  getLastDays().forEach((date: MomentDateOrString) => {
    if (typeof date === "string") {
      baseStatistics[date] = 0;
    }
    else {
      baseStatistics[date.format("MMM DD")] = 0;
    }
  });
  return { ...baseStatistics, ...doneStatistics };
};

export const mergeLanguageStatistics = (languageStatistics: ICount) => {
  const baseLanguageStats: ICount = {
    Bengali: 0,
    English: 0,
    Hindi: 0
  };
  return { ...baseLanguageStats, ...languageStatistics };
};

export const formatTimestamp = (value: string, item: any) => {
  const timestamp = _.get(item, value, false);
  return timestamp ? moment(timestamp).format("D.MM.YYYY") : "";
};
