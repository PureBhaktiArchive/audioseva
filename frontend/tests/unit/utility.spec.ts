import moment from "moment";
import { getDayDifference, getLastDays, mergeDoneStatistics, formatTimestamp } from "@/utility";

describe('utility', function () {
  beforeEach(() => {
    moment.now = () => +new Date();
  });

  test("getDayDifference", () => {
    moment.now = () => +new Date(1541497995699);
    const result = getDayDifference(1540630053685);
    expect(result).toEqual(10);
  });

  test("getLastDays", () => {
    moment.now = () => +new Date(1541497995699);
    const result = getLastDays();
    expect(result).toMatchSnapshot();
  });

  test("mergeDoneStatistics", () => {
    moment.now = () => +new Date(1541497995699);
    const doneStatistics = {
      today: 1,
      [moment().subtract(1, "days").format("MMM DD")]: 1
    };
    const results = mergeDoneStatistics(doneStatistics);
    expect(results).toMatchSnapshot();
  });

  test.only("formatTimestamp", () => {
    moment.now = () => +new Date(1541497995699);
    const item = {
      allotment: {
        timestampGiven: moment().valueOf()
      }
    };
    const results = formatTimestamp("allotment.timestampGiven", item);
    expect(results).toEqual("6.11.2018");
  })
});
