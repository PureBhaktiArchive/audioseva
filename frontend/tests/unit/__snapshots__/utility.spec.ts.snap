// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utility getLastDays 1`] = `
Array [
  "2018-11-01T08:53:15.699Z",
  "2018-11-02T08:53:15.699Z",
  "2018-11-03T08:53:15.699Z",
  "2018-11-04T09:53:15.699Z",
  "2018-11-05T09:53:15.699Z",
  "today",
]
`;

exports[`utility mergeDoneStatistics 1`] = `
Object {
  "Nov 01": 0,
  "Nov 02": 0,
  "Nov 03": 0,
  "Nov 04": 0,
  "Nov 05": 1,
  "today": 1,
}
`;
